miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -306,39 +314,72 @@ end
 
 #! Returns a boolean indicating whether the account is a regular immutable account.
 #!
-#! Inputs:  [acct_id]
+#! Inputs:  [acct_id_hi]
 #! Outputs: [is_immutable_account]
 #!
 #! Where:
-#! - acct_id is the account id.
+#! - acct_id_hi is the first felt of the account id.
 #! - is_immutable_account is a boolean indicating whether the account is a regular immutable
 #!   account.
 export.is_immutable_account
     exec.type push.REGULAR_ACCOUNT_IMMUTABLE_CODE eq
     # => [is_immutable_account]
 end
 
+#! Returns a boolean indicating whether the given account_ids are equal.
+#!
+#! Inputs:  [acct_id_hi, acct_id_lo, other_acct_id_hi, other_acct_id_lo]
+#! Outputs: [is_id_equal]
+#!
+#! Where:
+#! - acct_id_{hi,lo} are the first and second felt of an account id.
+#! - other_acct_id_{hi,lo} are the first and second felt of the other account id to compare against.
+#! - is_id_equal is a boolean indicating whether the account ids are equal.
+export.is_id_eq
+    movup.2 eq
+    # => [is_hi_equal, acct_id_lo, other_acct_id_lo]
+    movdn.2 eq
+    # => [is_lo_equal, is_hi_equal]
+    and
+    # => [is_id_equal]
+end
+
 #! Validates an account id.
 #!
-#! Inputs:  [acct_id]
+#! Inputs:  [account_id_hi, account_id_lo]
 #! Outputs: []
 #!
 #! Where:
-#! - acct_id is the account id.
+#! - account_id_{hi,lo} are the first and second felt of the account id.
 #!
 #! Panics if:
-#! - the account id is invalid: account id must have at least `MIN_ACCOUNT_ONES` ones.
+#! - account_id_hi does not contain version zero.
+#! - account_id_lo contains an anchor epoch that is greater or equal to 2^16.
+#! - account_id_lo does not have its lower 8 bits set to zero.
 export.validate_id
```

Comment:
I think naming can be improved here (i.e., we are not validating a full ID but rather some properties of an ID).